### PR TITLE
refactor: omega read from file

### PR DESCRIPTION
**Pull request by PrometheusPi**
In the current implementation the radiation plugin provides a method to set a path tp a list of frequency (omega) values to compute the radiation for. However, the path is set only via command line option and the value set via namespace setting is totally ignored. 

Since various parameters need to be set in `radiation.param` anyway, I removed the command line option (which is not doing anything if the list option has not be selected in `radiation.param` before) end set the path to the path defined in the namespace. 

The documentation was adjusted accordingly.

I would like to do a run-time test just to see whether anything else is not working properly with this frequency option before continuing to work on #800 (for the frequencies). 

- [x] run time test  

----------
**Update:**

With the latest version, all run time tests produced correct results. Both linear and logarithmic frequency scales work as before. The list now works again and has been tested with newlines, tabs and spaces as separators (in agreement with the newly updated documentation).

### CODE REVIEW

**Comment by ax3l** [comment]
I did not find more than a mention of `listLocation` in the `.rst`, did I overlook it? I though about adding docs on what's *inside* it (it's format) and where it ideally should reside in our inputs.

**Comment by PrometheusPi** [comment]
@ax3l Okay, I was referring to the single mention of `listLocation` [here](https://github.com/PrometheusPi/picongpu/blame/dc5954dec33e5a4a3921a7ff9b08e446cd53af29/docs/source/usage/plugins/radiation.rst#L61) which currently does not show up in the html version (fixed with #3052).

Since I am currently refactoring the entire frequency definition (with adding new docs) I think adding a (short-lived) longer documentation is not needed. But if you prefer a documentation, I will add another commit.

**Comment by PrometheusPi** [comment]
@ax3l #3052 does not yet add the planed changes. It just fixes a table formatting issue that prevented showing `listLocation` at all (in the html version).

**Comment by PrometheusPi** [comment]
I will add a commit to describe the currently needed setup.

File: share/picongpu/examples/Bunch/include/picongpu/param/radiation.param [hunk 1]
```diff
@@ -65,7 +65,8 @@ constexpr unsigned int N_omega = 2048; // number of frequencies
 
 namespace frequencies_from_list
 {
-//const char listLocation[] = "/scratch/s5960712/list001.freq";
+/** path to text file with frequencies */
+const char listLocation[] = "/path/to/frequency.list";
```
Comment (psychocoderHPC) [inline share/picongpu/examples/Bunch/include/picongpu/param/radiation.param]: Muste be `constexpr char listLocation[]` to fit the documentation above. If this make compile issues change it to `constexpr char const listLocation* = "path";`

File: share/picongpu/examples/KelvinHelmholtz/include/picongpu/param/radiation.param [hunk 2]
```diff
@@ -68,7 +68,8 @@ constexpr unsigned int N_omega = 1024; // number of frequencies
 
 namespace frequencies_from_list
 {
-//const char listLocation[] = "/scratch/s5960712/list001.freq";
+/** path to text file with frequencies */
+const char listLocation[] = "/path/to/frequency.list";
```
Comment (psychocoderHPC) [inline share/picongpu/examples/KelvinHelmholtz/include/picongpu/param/radiation.param]: same here please sue `constexpr`

File: share/picongpu/examples/Bunch/include/picongpu/param/radiation.param (hunk 1 above)
Comment (sbastrakov) [reply share/picongpu/examples/Bunch/include/picongpu/param/radiation.param]: I think actually `constexpr const char *` is preferable over `constexpr char listLocation[]` due to more operations allowed with such a variable w/o requiring a definition.

**Comment by PrometheusPi** [comment]
Currently, using the omega by list system fails with a segfault. I will thus put this pull request as **WIP**. This seems to be not caused by the frequency reading - the vernose autput confirms that data has been read correctly. I suspect an outdated use of the data box. @psychocoderHPC do you see anything obviously wrong. 
In the next test I will add debug options and post more results of the failing here.

**Comment by psychocoderHPC** [comment]
> Currently, using the omega by list system fails with a segfault. I will thus put this pull request as **WIP**. This seems to be not caused by the frequency reading - the verbose output confirms that data has been read correctly. I suspect an outdated use of the data box. @psychocoderHPC do you see anything obviously wrong.
> In the next test I will add debug options and post more results of the failing here.

We found out @PrometheusPi accesses device memory from the host side and therefore get the segfault

**Comment by PrometheusPi** [comment]
@psychocoderHPC all fixes are included now - can be reviewed

File: docs/source/usage/plugins/radiation.rst [hunk 3]
```diff
@@ -62,6 +62,28 @@ namespace                     Description
...
+The number of total sample frequencies ``N_omega`` need to be defined as ``constexpr unsigned int``.
+In the sub-namespace ``SI``, a minimal frequency ``omega_min`` and a maximum frequency ``omega_max`` need to be defined as ``constexpr float_64``.
+
+For the **logarithmic frequency** scale all definitions need to be in the ``picongpu::plugins::radiation::log_frequencies`` namespace. 
+Equivalently to the linear case, three variables need to be defined: 
+The number of total sample frequencies ``N_omega`` need to be defined as ``constexpr unsigned int``.
+In the sub-namespace ``SI``, a minimal frequency ``omega_min`` and a maximum frequency ``omega_max`` need to be defined as ``constexpr float_64``.
+
+For the **file-based frequency** definition,  all definitions need to be in the ``picongpu::plugins::radiation::frequencies_from_list`` namespace.
+The number of total frequencies ``N_omega`` need to be defined as ``constexpr unsigned int``  and the path to the file containing the frequency values un units of :math:`[s^{-1}]` needs to be given as ``constexpr const char * listLocation = "/path/to/frequency_list";``.
```
Comment (psychocoderHPC) [inline docs/source/usage/plugins/radiation.rst]: `... values un units ...` should be `...  and units ...`

File: docs/source/usage/plugins/radiation.rst [hunk 4]
```diff
@@ -62,6 +62,28 @@ namespace                     Description
...
+The frequency values in the file can be separated by newlines, spaces, tabs, or any other whitespace. The numbers should be given in such a way, that c++ standard ``std::ifstream`` can interpret the number e.g., as ``2.5344e+16``. 
+
+.. note::
+
+   Currently, the variable ``listLocation`` is required to be defined in the ``picongpu::plugins::radiation::frequencies_from_list`` namespace.
```
Comment (psychocoderHPC) [inline docs/source/usage/plugins/radiation.rst]: Please extent with `... even if frequencies_from_list is not used`. This is in the current documentation not fully clear.

File: docs/source/usage/plugins/radiation.rst (hunk 3 above)
Comment (PrometheusPi) [reply docs/source/usage/plugins/radiation.rst]: great catch 👍 should be `in units`

**Comment by PrometheusPi** [comment]
@psychocoderHPC The plan was not to convert to `struct` right away but to first of all fix the bug in frequencies from list in a separate pull request - then work on the refactoring of the interface. Thus I did not use `struct`s right away, but will add them in a followup pull request.

**Comment by PrometheusPi** [comment]
@psychocoderHPC I added your suggestions (except `struct`- see comment before on that)

**Comment by psychocoderHPC** [comment]
Ok I am fine with fixing the bug and adding documentation only.

File: docs/source/usage/plugins/radiation.rst (hunk 3 above)
Comment (psychocoderHPC) [reply docs/source/usage/plugins/radiation.rst]: saw that @ax3l  allready mentioned it.

**Pull request merged**